### PR TITLE
Feat/tooltip/refactor

### PR DESCRIPTION
**Pull request by rayyan224**

### CODE REVIEW

File: src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -1,24 +1,63 @@
...
+    const popoverRef = useRef(null);
+    const [width, setWidth] = useState(0);
+    const [height, setHeight] = useState(0);
+    const [popverWidth, setPopoverWidth] = useState(0);
+    const [popoverHeight, setPopoverHeight] = useState(0);
+    useEffect(() => {
+        console.log(
+            'width',
+            parretnRef.current ? (parretnRef.current as any).offsetWidth : 0,
+        );
```
Comment (locothedev) [inline src/components/Tooltip/Tooltip.tsx:26]: console.log no bueno

File: src/components/Tooltip/types.ts [hunk 2]
```diff
@@ -22,7 +17,7 @@ export interface TooltipProps {
     position: Position;
 
     /**
-     * Set text to show in tooltip
+     * Set element/text that will be shown within the tooltip
      */
-    text?: string;
+    content?: JSX.Element | string;
```
Comment (locothedev) [inline src/components/Tooltip/types.ts:22]: lets make content required? What do you think about it

File: src/components/Tooltip/Tooltip.tsx [hunk 3]
```diff
@@ -1,24 +1,63 @@
...
-            position={position}
-            data-tooltip={text}
+        <StyledTooltipParentDiv
+            ref={parretnRef}
             data-testid={'tooltip-container-test-id'}
-            maxWidth={maxWidth}
         >
-            <div data-testid={'tooltip-children-test-id'}>{children}</div>
-        </TooltipContainer>
+            <span>{children}</span>
```
Comment (locothedev) [inline src/components/Tooltip/Tooltip.tsx:45]: as we are allowing to pass in jsx elements I dont think its good allowing to pass in `div` inside `span`.

What is the actual difference between children and content?

**Review by locothedev — CHANGES_REQUESTED** [review]
Awesome bro 👍

File: src/components/Tooltip/types.ts (hunk 2 above)
Comment (rayyan224) [reply src/components/Tooltip/types.ts:22]: Yea, ur right

File: src/components/Tooltip/Tooltip.tsx (hunk 3 above)
Comment (rayyan224) [reply src/components/Tooltip/Tooltip.tsx:45]: Children is for what is shown 
content is what's shown in the tooltip

**Pull request merged**